interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/internal/propertyAccessors.scala

```diff
@@ -162,14 +162,18 @@ object propertyAccessors {
   }
 
   object MapPropertyAccessor extends PropertyAccessor with ReadOnly {
-
+    // if map does not contain property this should return false so that `NoParamMethodPropertyAccessor` can be used
     override def canRead(context: EvaluationContext, target: scala.Any, name: String): Boolean =
-      true
+      target.asInstanceOf[java.util.Map[_, _]].containsKey(name)
```

Comment:
Don't really know what to do with this. If this always returns true then we can never call no-parameter methods on records like `empty`  - I think one solution would be for this to return false if the record doesn't have a given key - then the spel parser looks for another accessor - it finds a no-parameter method through `NoParamMethodPropertyAccessor` and uses that. 

But if it returns true only if it contains given key then for example in case we have a typed record with a field `foo` and in runtime the record does not have that key then we get a spel exception - not a `null` like previously. I think a `null` in that case is desired behaviour - there is a test for this - https://github.com/TouK/nussknacker/blob/cac675ce8518f828393c567e05d82055ce86296b/interpreter/src/test/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSpec.scala#L591-L614. 

Don't see any simple solution and don't know if I've explained it clearly - would gladly talk it over offline :smiley: